### ISSUE TITLE
Add the Ability to Transform Condition Values After the "when()" Check 

### ISSUE THREAD

**Issue by pky920216344** [mybatis/mybatis-dynamic-sql#105]
example:
![image](https://user-images.githubusercontent.com/23168199/60523004-8b1ba280-9d1c-11e9-92b9-8e860b26c0dd.png)
![image](https://user-images.githubusercontent.com/23168199/60523019-92db4700-9d1c-11e9-805a-78d65137967f.png)
 like this, parameter is null, but "%"+parameter+"%" is not null
When do fuzzy query operation, i must coding:
if(null!=parameter){
  xxx.and(column, ()-> "%"+parameter+"%");
}
it's not  beautiful

**Comment by jeffgbutler** [comment]
@pky920216344 Wow - this is a great idea - thanks!

I would implement it more like this:

```java
public IsLike<T> then(UnaryOperator<T> transformer) {
    return shouldRender() ? new IsLike<>(() -> transformer.apply(value())) : this;
}
```

Then we could write code like this:

```java
SelectStatementProvider selectStatement = select(id, firstName, lastName)
        .from(person)
        .where(firstName, isLike(fName).when(Objects::nonNull).then(s -> "%" + s + "%"))
        .and(lastName, isLike(lName).when(Objects::nonNull).then(s -> "%" + s + "%"))
        .build()
        .render(RenderingStrategy.MYBATIS3);
```

What do you think?

**Comment by jeffgbutler** [comment]
If you write a little utility class like this:

```java
public class SearchUtils {
    public static String addWildcards(String s) {
        return "%" + s + "%";
    }
}
```

Then you can write code like this:

```java
SelectStatementProvider selectStatement = select(id, firstName, lastName)
    .from(person)
    .where(firstName, isLike(fName).when(Objects::nonNull).then(SearchUtils::addWildcards))
    .and(lastName, isLike(lName).when(Objects::nonNull).then(SearchUtils::addWildcards))
    .build()
    .render(RenderingStrategy.MYBATIS3);
```

I think this qualifies as beautiful 😄

**Comment by pky920216344** [comment]
@jeffgbutler  better than me  hope I can use it as soon as possible 😀

**Comment by jeffgbutler** [comment]
I’ll add this and do a release very soon.

**Comment by jeffgbutler** [comment]
@pky920216344 this is released and available now in version 1.1.2

**Comment by pky920216344** [comment]
@jeffgbutler Thank you 😄
